01_slides/05_strings.ipynb

```diff
@@ -609,6 +613,256 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "6cee9074",
+      "metadata": {},
+      "outputs": [
+        {
+          "data": {
+            "text/plain": [
+              "'I AM NOT YELLING'"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "# convert a string to all caps\n",
+        "'I am not yelling'.upper()"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "2c7eafdf",
+      "metadata": {},
+      "outputs": [
+        {
+          "data": {
+            "text/plain": [
+              "0"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "# count the 'e's\n",
+        "'This string is unusual'.count('e')"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "9e016ebe",
+      "metadata": {},
+      "outputs": [
+        {
+          "data": {
+            "text/plain": [
+              "True"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "# check if a string ends with a substring\n",
+        "'file_name.csv'.endswith('.csv')"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "a9beef6b",
+      "metadata": {},
+      "outputs": [
+        {
+          "data": {
+            "text/plain": [
+              "'long_file_name_with_spaces.csv'"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "# replace text\n",
+        "'long file name with spaces.csv'.replace(' ', '_')"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "9fefba25",
+      "metadata": {},
+      "source": [
+        "For a full list of string methods, see the [Python documentation](https://docs.python.org/3/library/stdtypes.html#string-methods)."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "5e237cbd",
+      "metadata": {},
+      "source": [
+        "### Formatting Strings\n",
+        "\n",
+        "The `format()` string method makes it easier to combine text with other data. We can create text templates with curly braces (`{}`), then pass values in."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "5ae0487a",
+      "metadata": {},
+      "outputs": [
+        {
+          "data": {
+            "text/plain": [
+              "\"Ada Lovelace's initials are A. L.\""
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "'Ada Lovelace\\'s initials are {}. {}.'.format('Ada'[0], 'Lovelace'[0])"
```

Comment:
This is a new syntax learners haven't been introduced to before. 

Previously it was:
```
first_name = "Ada"
print(first name[0])
```

Now we're showcasing it is possible to do string indexing directly:

```
first_letter = "Ada"[0]
```

I'm not against it, just wondering if this is a jump we should consider.